src/FolderContext.ts

```diff
@@ -0,0 +1,50 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VSCode Swift open source project
+//
+// Copyright (c) 2021 the VSCode Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VSCode Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as vscode from 'vscode';
+import { PackageWatcher } from './PackageWatcher';
+import { SwiftPackage } from './SwiftPackage';
+
+export class FolderContext implements vscode.Disposable {
+    private packageWatcher?: PackageWatcher;
+
+	private constructor(
```

Comment:
Inconsistent formatting
```suggestion
    private constructor(
```